src/test/regress/expected/locally_execute_intermediate_results.out

```diff
@@ -142,8 +145,10 @@ HAVING max(value) > (SELECT max FROM cte_1 JOIN cte_2 USING (max));
 DEBUG:  generating subplan XXX_1 for CTE cte_1: SELECT max(value) AS max FROM locally_execute_intermediate_results.table_1
 DEBUG:  generating subplan XXX_2 for CTE cte_2: SELECT max(value) AS max FROM locally_execute_intermediate_results.table_1
 DEBUG:  Plan XXX query after replacing subqueries and CTEs: SELECT count(*) AS count FROM locally_execute_intermediate_results.table_2 GROUP BY key HAVING (max(value) OPERATOR(pg_catalog.>) (SELECT cte_1.max FROM ((SELECT intermediate_result.max FROM read_intermediate_result('XXX_1'::text, 'binary'::citus_copy_format) intermediate_result(max text)) cte_1 JOIN (SELECT intermediate_result.max FROM read_intermediate_result('XXX_2'::text, 'binary'::citus_copy_format) intermediate_result(max text)) cte_2 USING (max))))
-DEBUG:  Subplan XXX_1 will be written to local file
-DEBUG:  Subplan XXX_2 will be written to local file
+DEBUG:  Subplan XXX_1 will be sent to localhost:xxxxx
```

Comment:
As you noted, I think it might be useful to adjust these tests to write subplans locally. But, these tests are also now useful as they verify HAVING pushdown works fine. Maybe have additional tests  with GROUP BY non-dist-key  or such?

Though, very interesting performance trade-off, we need to move more data here.

Reply:
changed to group by non-distribution column as appropriate based on the description of the test, or duplicated the test to cover both cases